src/jtag3.c

```diff
@@ -1624,7 +1649,7 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
   return rv;
 }
 
-int jtag3_open_common(PROGRAMMER *pgm, const char *port) {
+int jtag3_open_common(PROGRAMMER *pgm, const char *port, const int mode_switch) {
```

Comment:
Please remove the const in const int mode_switch. It serves little purpose: the compiler passes a *copy* of the int argument to the function, so the caller does not need to worry what the function does with that copy. Some style guides discourage the use of const for call-by-value variables, and AVRDUDE is not in the habit of consting int aruments, so I wouldn't start that here.

The other const in const char *port is important, though, as it warrants that the function does not try to modify what port points at (call by reference).

Reply:
Thanks for the tip!